### DISCUSSION TITLE
Independent encode of radius and theta

### DISCUSSION THREAD

**Discussion by qnzhou** [vega/altair#3204] [Q&A] [answered]
Hi there,

I am playing with [this example](https://altair-viz.github.io/gallery/radial_chart.html) using arc as mark. While the code works well as it is, however, if I assign radius to another column in the data frame, the plot looks completely wrong. Here is an example:

```python
source = pd.DataFrame({"values": [12, 23, 47, 6, 52, 19]})
source["radius"] = source.values # <-- Create a new column for radius.

base = alt.Chart(source).encode(
    alt.Theta("values:Q").stack(True),
    alt.Radius("radius:Q").scale(type="sqrt", zero=True, rangeMin=20),
    color="values:N",
)

base.mark_arc(innerRadius=20, stroke="#fff")
```
Here is what I get:
![image](https://github.com/altair-viz/altair/assets/3606672/d6f399a1-ead7-4671-af1c-c3c505393da0)

I am using the latest atlair version 5.1.1. Any clues on why it is the case?

**Comment by joelostblom** [comment]
Hmm, not sure what happens there, but it is related to how Vega-Lite renders the chart. It works if you set both columns to `'radius'` too. You can report it in the Vega-Lite repo by clicking the three dots in the top right corner and creating a link to the sped in the Vega Editor

**Reply by qnzhou** [reply]
With opacity turned to 0.5, it reveals more info about the chart. I think the `stack=True` setting for theta channel is not respected.
<img width="411" alt="image" src="https://github.com/altair-viz/altair/assets/3606672/5d32c506-00f5-4643-b004-a69093fc417e">

Digging a bit deeper into the compiled vega-lite json setting, I see the following stack transform:
```json
        {
          "type": "stack",
          "groupby": ["radius"],
          "field": "values",
          "sort": {"field": ["values"], "order": ["ascending"]},
          "as": ["values_start", "values_end"],
          "offset": "zero"
        },
```

Note that it contains a `groupby` setting that is set to `["radius"]` (instead of `[]` when both radius and theta uses the same data), which seems incorrect. This also shows the problem is probably with Altair instead of Vega-Lite.

**Reply by joelostblom** [reply]
Nice find digging deeper! Since the issue occurs in the Compiled Vega spec, it means something is likely wrong in the translation of the Vega-Lite spec to the Vega spec. This happens on the Vega-Lite side of things, Altair only renders Vega-Lite specification, but does not control how this spec is translated into a Vega spec

**Reply by qnzhou** [reply, marked as answer]
Thanks! It seems this bug has been [reported back Feb 2022](https://github.com/vega/vega-lite/issues/7957) but has not been fixed in Vega-Lite. This bug unfortunately makes altair/vega-lite incapable of creating radial chart.

**Reply by joelostblom** [reply]
Thanks for digging up that issue. I will let the discussion continue there ~and close this one~ since there is nothing we can do on the Altair side of things to fix this.

Edit: No need to close this I guess since it is a discussion and not an issue, but I marked your answer as the "solution"

**Reply by hydrosquall** [reply]
Good news is that a graph can be drawn correctly with regular `Vega` (here's a [proof of concept](https://github.com/vega/vega-lite/issues/7957#issuecomment-2601380930))! We will fix it in Vega-Lite too but it may take a bit longer to get that released.

In the meantime, can Altair let you pass in a plain Vega specification, or is it limited to rendering Vega-Lite specs?
